### DISCUSSION TITLE
"Process stages" schema

### DISCUSSION THREAD

**Discussion by atherdon** [atherdon/dash-back#26] [General]
думаю что stages у нас может быть много, я легко готов расширить из текущих 6 в 10-12 например.
прикрепляю документ, в котором мы обсуждаем добавление нескольких stages - связанных с фильтрацией спама

[stages.pdf](https://github.com/atherdon/dash-back/files/5946815/stages.pdf)

**Comment by kolserdav** [comment]
Первое что пришло на ум.
Добавить матералу поле `editing` {boolean} если 1 то кнопка редактировать не активна и появляется кнопка подписаться. Когда тот кто нажал редактировать, сохраняет изменения, `editing` ставится в 0 и запускается таймер на по истечению которого, если `editing` равно 0 делается рассылка подписавшимся. Подписавшихся вообще просто в памяти хранить и поcле рассылки чистить. Единственное, чтобы при нажатии на редактировать не было ошибок (если открыли материал примерно в одно время, но один нажал раньше на редатировать), то отслеживать её состояние по `ws`.

**Reply by kolserdav** [reply]
Подумал. Тут проблема будет если нажмет редактировать и после не нажмет сохранить, а материал так и будет editing. Там может ккакой нибудь `onbeforeunload` навесить, но я как-то пытался он не работал в новых версиях браузера

**Reply by atherdon** [reply]
Вначале мы будем менять эти статусы руками. плюс, именно проблема спама будет частично решена, т.к. помощник редактора ручками будет копировать данные из основной дашбоард. но когда мы сделаем sync с основной базой - спам опять польется.

насчет статусов - я думаю что у нас будет дропдаун, который просто будет или двигать "вперед" по статусам(например submitted -> not spam -> edited -> title -> scheduled -> published) или можно будет просто перескочить. кстати эту часть sync можно раньше попросить наших девов сделать. 
просто ужно чтобы они подрубились к нашему серверу, в тот момент когда происхоит например event publish.

я предоставлю больше инфы чуть позже. просто хотел стартануть дискуссию о создании реальной схемы, которую можно будет потом разрабатывать на фронте, когда мы все запустим

**Comment by atherdon** [comment]
We can steal this schema for our stages: https://stackoverflow.com/a/40845966

**Reply by kolserdav** [reply]
I don't understand.  ?

**Reply by atherdon** [reply]
For me - logic of editorial stages is similar to Kanban board. Stackoverflow answer containing DB table that can keep information related to articles and their stages. We can google more about it. Just don't want to re-invent the wheel here.

**Comment by atherdon** [comment]
Stages\statuses that we have right now. But we probably will have more than 11 in the future. Will post more details later
 
![Story Statuses 2](https://user-images.githubusercontent.com/1469198/107992722-f42f5700-6fe1-11eb-9748-a5f642a77ebf.png)

**Comment by atherdon** [comment]
Я смог придумать, как разделить statuses & stages. Это также позволит не иметь 11 шагов в нашей админке.

- Articles will have statuses.
- Editorial process will have stages.
So it's a different entities, with mixed things that can be separated.
